HIP/hip-514-missing-hts-precompile-functions.md

```diff
@@ -0,0 +1,244 @@
+| `TokenInfo tokenInfo`     | The shared hedera token info                                                                  |
+| `uint64 serialNumber`     | The serial number of the nft                                                                  |
+| `address ownerId`         | The account id specifying the owner of the non fungible token                                 |
+| `uint32 creationTime`     | The epoch second at which the token was created.                                              |
+| `bytes metadata`          | The unique metadata of the NFT                                                                |
+| `address spenderId`       | The account id specifying an account that has been granted spending permissions on this nft   |
+| `string ledgerId`         | The ID of the network ledger                                                                  |
+
+### Solidity Function Signatures
+
+The ABI signature and hashes for token management functions are as follows:
+
+| hash        | effective signature                               | return                            |
+| ------------|---------------------------------------------------|-----------------------------------|
+| `927da105`  | `allowance(address, address, address)`            | `(int256, uint256)`               |
+| `e1f21c67`  | `approve(address, address, uint256)`              | `(int256)`                        |
+| `7336aaf0`  | `approveNFT(address, address, uint256)`           | `(int256)`                        |
+| `098f2366`  | `getApproved(address, uint256)`                   | `(int256, address)`               |
+| `f49f40db`  | `isApprovedForAll(address, address, address)`     | `(int256, bool)`                  |
+| `367605ca`  | `setApprovalForAll(address, address, bool)`       | `(int256)`                        |
+| `46de0fb1`  | `isFrozen(address, address)`                      | `(int256, bool)`                  |
+| `f2c31ff4`  | `isKyc(address, address)`                         | `(int256, bool)`                  |
+| `f069f712`  | `deleteToken(address)`                            | `(int256)`                        |
+| `ae7611a0`  | `getTokenCustomFees(address)`                     | `(int256, CustomFee[])`           |
+| `a7daa18d`  | `getTokenDefaultFreezeStatus(address)`            | `(int256, bool)`                  |
+| `335e04c1`  | `getTokenDefaultKycStatus(address)`               | `(int256, bool)`                  |
+| `d614cdb8`  | `getTokenExpiryInfo(address)`                     | `(int256, Expiry)`                |
+| `3f28a19b`  | `getFungibleTokenInfo(address)`                   | `(int256, FungibleTokenInfo)`     |
+| `1f69565f`  | `getTokenInfo(address)`                           | `(int256, TokenInfo)`             |
+| `3c4dd32e`  | `getTokenKey(address, uint)`                      | `(int256, KeyValue)`              |
+| `2c20dcd1`  | `getNonFungibleTokenInfo(address, uint32)`        | `(int256, NonFungibleTokenInfo)`  |
```

Comment:
For the serial number I think we should use `int64` instead of `uint32`, since in `mint` we return `int64[]` for the new serial numbers.

Reply:
Fixed to make int64 in all cases